packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -532,24 +544,31 @@ export class CfnParser {
       return leftHalf + CfnReference.for(refResource, attribute, true).toString() + this.parseFnSubString(rightHalf, map);
     }
   }
-}
 
-function specialCaseRefs(value: any): any {
-  switch (value) {
-    case 'AWS::AccountId': return Aws.ACCOUNT_ID;
-    case 'AWS::Region': return Aws.REGION;
-    case 'AWS::Partition': return Aws.PARTITION;
-    case 'AWS::URLSuffix': return Aws.URL_SUFFIX;
-    case 'AWS::NotificationARNs': return Aws.NOTIFICATION_ARNS;
-    case 'AWS::StackId': return Aws.STACK_ID;
-    case 'AWS::StackName': return Aws.STACK_NAME;
-    case 'AWS::NoValue': return Aws.NO_VALUE;
-    default: return undefined;
+  specialCaseSubRefs(value: string, pseudoParameter: boolean, refElement?: CfnElement ): string | undefined {
+    if (pseudoParameter) {
+      return value.indexOf('::') === -1 ? undefined: '${' + value + '}';
+    }
+    return value in (this.parameters) ? this.parameters[value] : CfnReference.for(refElement!, 'Ref', true).toString();
```

Comment:
I don't understand this code, sorry. In my mind, this is how this method should look like:

```ts
private specialCaseSubRefs(value: string ): string | undefined {
  if (value in this.parameters) {
   return this.parameters[value];
  } else {
    return value.indexOf('::') === -1 ? undefined: '${' + value + '}';
  }
```

No `pseudoParameter` argument, no `refElement` argument.